pallets/pooled-staking/src/tests.rs

```diff
@@ -0,0 +1,664 @@
+            assert_eq!(pool_before.hold + balance, pool_after.hold);
+            assert_eq!(before.delegator_balance + balance, after.delegator_balance);
+            assert_eq!(before.staking_balance - balance, after.staking_balance);
+        }
+        SignedBalance::Negative(balance) => {
+            assert_eq!(pool_before.hold - balance, pool_after.hold);
+            assert_eq!(before.delegator_balance - balance, after.delegator_balance);
+            assert_eq!(before.staking_balance + balance, after.staking_balance);
+        }
+    }
+
+    // Stake stay the same.
+    assert_fields_eq!(pool_before, pool_after, stake);
+}
+
+pool_test!(
+    fn empty_delegation<P>() {
+        ExtBuilder::default().build().execute_with(|| {
+            let before = State::extract(ACCOUNT_CANDIDATE_1, ACCOUNT_DELEGATOR_1);
+            let pool_before =
+                PoolState::extract::<Joining>(ACCOUNT_CANDIDATE_1, ACCOUNT_DELEGATOR_1);
+
+            assert_noop!(
+                Staking::request_delegate(
+                    RuntimeOrigin::signed(ACCOUNT_DELEGATOR_1),
+                    ACCOUNT_CANDIDATE_1,
+                    P::target_pool(),
+                    0
+                ),
+                Error::<Runtime>::StakeMustBeNonZero
+            );
+
+            let after = State::extract(ACCOUNT_CANDIDATE_1, ACCOUNT_DELEGATOR_1);
+            let pool_after =
+                PoolState::extract::<Joining>(ACCOUNT_CANDIDATE_1, ACCOUNT_DELEGATOR_1);
+
+            assert_eq!(before, after);
+            assert_eq!(pool_before, pool_after);
+
+            assert_eq_events!(Vec::<Event<Runtime>>::new());
+        })
+    }
+);
+
+pool_test!(
+    fn delegation_request<P>() {
+        ExtBuilder::default().build().execute_with(|| {
+            let amount = 3324;
+            do_request_delegation(
+                ACCOUNT_CANDIDATE_1,
+                ACCOUNT_DELEGATOR_1,
+                P::target_pool(),
+                amount + 1, // to test joining rounding
+                amount,
+            );
+
+            assert_eq_events!(vec![
+                Event::IncreasedStake {
+                    candidate: ACCOUNT_CANDIDATE_1,
+                    stake: amount,
+                },
+                Event::UpdatedCandidatePosition {
+                    candidate: ACCOUNT_CANDIDATE_1,
+                    stake: amount,
+                    self_delegation: 0,
+                    before: None,
+                    after: None,
+                },
+                Event::RequestedDelegate {
+                    candidate: ACCOUNT_CANDIDATE_1,
+                    delegator: ACCOUNT_DELEGATOR_1,
+                    towards: P::target_pool(),
+                    pending: amount
+                },
+            ]);
+        })
+    }
+);
+
+pool_test!(
+    fn delegation_request_more_than_available<P>() {
+        ExtBuilder::default().build().execute_with(|| {
+            let amount = DEFAULT_BALANCE + 1;
```

Comment:
If I change this to `DEFAULT_BALANCE` it also says "FundsUnavailable", I guess because of existential deposit being set to 1. Is that expected? `DEFAULT_BALANCE - 1` works as expected.